runtime/lua/vim/_comment.lua

```diff
@@ -3,6 +3,44 @@
 ---@field left string Left part of comment
 ---@field right string Right part of comment
 
+---@param parser vim.treesitter.LanguageTree
+---@param row integer
+---@param col integer
+---@return string?
+local function get_metadata_commentstring(parser, row, col)
+  local bufnr = vim.api.nvim_get_current_buf()
+  local highlighter = vim.treesitter.highlighter.active[bufnr]
+  if not highlighter then
+    return
+  end
+  local query = highlighter:get_query(parser:lang()):query()
+  if not query then
+    return
+  end
+  local range = { row, col, row, col }
+  local tree = assert(parser:tree_for_range(range))
+  local last_node = tree:root()
+  ---@type string?
+  local commentstring
+  for _, match, metadata in query:iter_matches(tree:root(), bufnr, row, row + 1) do
```

Comment:
All this tree-sitter magic is above my head, so can't really provide much useful feedback here.

One thing I'd expect is that all the tree-sitter routing might be a good candidate (or already is) for some `vim.treesitter` function. Can one of `vim.treesitter.get_captures_at_pos()` or `vim.treesitter.get_node()` be used here somehow?

Reply:
Nice thanks! `get_captures_at_pos()` worked perfectly- note to self, update `_get_urls()` to use the same...